sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -344,8 +344,10 @@ public void finish(@Nullable SpanStatus status) {
 
       if (timer != null) {
         synchronized (timerLock) {
-          timer.cancel();
-          timer = null;
+          if (timer != null) {
```

Comment:
Double check lock, in order to not lock if timer is already null. Looks like a good pattern to me.

Reply:
Yes, as @brustolin said, it's there to avoid the lock if it isn't needed.

@marandaneto created https://github.com/getsentry/sentry-java/issues/2202 to track the double checked locking improvement.

Reply:
The double lock is a bad pattern in Java and is [broken by default](https://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html) @brustolin 
In this case, it works, since you made the field `volatile` as well, and the first lock is just for the contention

Reply:
That seems specific to the JVM. Does it affect ART? Also, there are some 'theoreticals' in there (I just skimmed though)
Also, we're not initializing the Timer in this method, so what he's talking about doesn't seem to apply here.

Reply:
The shared article is more about how you can do the double-checked locking wrong, and that's why it's considered an anti-pattern, the example indeed isn't the same, but I shared it just for context since @brustolin mentioned "good pattern".

> In this case, it works, since you made the field volatile as well

Reply:
https://stackoverflow.com/questions/5717090/double-checked-locking-in-android